x/bank/keeper/msg_server.go

```diff
@@ -15,6 +14,17 @@ type msgServer struct {
 	Keeper
 }
 
+func (k msgServer) UpdateDenomMetadata(goCtx context.Context, msg *types.MsgUpdateDenomMetadata) (*types.MsgUpdateDenomMetadataResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	k.Keeper.SetDenomMetaData(ctx, *msg.Metadata)
+
+	//TODO: telemetry?
+
+	// TODO: should it emit events?
```

Comment:
Telemetry can be skipped. I don't think there needs to be any events emitted either.